packages/swing-store/src/snapStore.js

```diff
@@ -81,10 +87,11 @@ export function makeSnapStore(
       compressedSize INTEGER,
       compressedSnapshot BLOB,
       PRIMARY KEY (vatID, snapPos),
-      UNIQUE (vatID, inUse),
-      CHECK(compressedSnapshot is not null or inUse is null)
+      UNIQUE (vatID, inUse)
```

Comment:
Oh rats, I just realized this change (removing the CHECK) means this PR won't work across an upgrade. I needed to remove it to allow the import process to create the record without a snapshot first, then populate the `compressedSnapshot` column a moment later.

I need to think of a way to modify this table at startup, if the check is present, or to use `compressedSnapshot = ""` or `0` or something to represent "not populated".

Reply:
Or.. maybe this isn't a significant problem. The mainnet chain contains swing-store version 0.9.1. The only existing mainnet DBs were created by one of two pathways:
* `agoric-upgrade-10`, which moved to `mainnet1B` and does an `initSwingStore`
* a state-sync import using `mainnet1B`, which does the old (lossy) `importSwingStore`
both cases will get you a DB with the `CHECK(compressedSnapshot is not null or inUse is null)` constraint.

Let's say the version we're building here is named `0.10.0`, and is deployed to mainnet via `agoric-upgrade-11`. Those two pathways will cause:
* `agoric-upgrade-11` will leave the 0.9.1 schema in place
* state-sync import will create a DB with the 0.10.0 schema (lacking the CHECK)

So it's ok for `importSwingStore` to violate the old CHECK constraint, because that code will only encounter a fresh DB, with the 0.10.0 schema.

The normal runtime `snapStore.saveSnapshot()` function might encounter either schema, so it must obey the CHECK constraint. I didn't change that method, so it's still using the one-statement `sqlSaveSnapshot` operation, which populates `compressedSnapshot` immediately.

We'll wind up with some mainnet instances with one schema, and some with the other.

Either we need to make this code do an ALTER TABLE to remove the CHECK constraint from everybody, so they all wind up with the same schema, or we need to safely tolerate either schema going forward.

Hm.. how should I write a test that makes sure we don't accidentally change `saveSnapshot` in the future? Maybe a test which does `initSwingStore`, then manually doe an `ALTER TABLE` to add the CHECK that just got removed, then run the normal `test-snapstore.js` tests?

Reply:
Please see https://github.com/Agoric/agoric-sdk/pull/7635 as it requires removing the constraint as well.

Reply:
I'm going to sketch out the process we'd use for schema migration, but not implement it for this release. We'll need to implement it before landing #7635, or before making other changes to the way we use the table, but I'd like to avoid the churn/risk of doing it now (especially because of the more extensive testing we need to build to have confidence in it). See https://github.com/Agoric/agoric-sdk/issues/8089 for details.